src/core/apolloClient/cache.ts

```diff
@@ -66,6 +66,9 @@ export const cache = new InMemoryCache({
         customerCreditNotes: {
           keyArgs: ['invoiceId'],
         },
+        customerMetadata: {
```

Comment:
I'm confused... Why do you need to do that ?

Reply:
agree, did that cause of a cache warning asking to do so somehow.

Will try again to understand what could cause it

Reply:
Ok so here is what happens

## Context

We have an apollo warning when deleting metadata and saving a customer with those changes.

Reason is that apollo does not know how to reconcile this array of objects that have theyr own type.
Specially because when deleting the items from the UI, We delete them from the Formik values but not from the cache.
So when we udpate the customer, we have the current cached version of the customer containing metadata that does not exists anymore.

## Explorations

I tried to 
- manually evict deleted elements
- define a customer merge function for metadata in `cache.ts`
- remove the manual udpate cache method and try toi mix fragment definition to attempt to make apollo understand by itself which object to update

None of those worked. The interesting about the last item is that 
- The customerDetails and Customer types are somehow mixed in the app. We should only have one type imo
- This part of the app is pretty message in term of fragment definition. There is a wrong pattern where data is asked in parent and spread over all children, and also a hook is taking care or defining expected data on mutation result, but it looks to not match ideally with the fragments for the list / details of customer.

## Next steps

We need to
1.  clean types definition on BE side and reflect this change on FE
2. clear fragments definition and data fetching hierarchy in the app

I'll create a ticket for this on FE repo as in our Notion kanban and take care of both BE and FE 

Also, I'll resolve this conversation and keep the change as it is in this PR as it's the less ugly solution I would say